File: testing-compiler-interface/src/main/java/scala/scalanative/api/CompilerError.java
package scala.scalanative.api;

public interface CompilerError {

    /**
     * Returns the point of the position where the error is reported.
     * Note a single point is returned even if range positions are enabled.
     *
     * @return point of the position
     */
    public Integer getPosition();

    /**
     * Returns the message of the error, as reported by the compiler.
     *
     * @return String value representing the error message
     */
    public String getErrorMsg();

}

File: testing-compiler/src/main/scala/scalanative/compiler/reporter/CompilerError.scala
package scala.scalanative.compiler.reporter

import scala.scalanative.api

private[scalanative] case class CompilerError(pos: Int, msg: String)
    extends api.CompilerError {
  override def getPosition: Integer = pos

  override def getErrorMsg: String = msg

  override def equals(obj: Any): Boolean = obj match {
    case err: api.CompilerError =>
      (err.getErrorMsg == getErrorMsg) &&
        (err.getPosition == getPosition)
    case _ =>
      false
  }
}
